### PR TITLE
Support the “minimal” TCF experience API and dynamically fetch “full” API response after banner is shown

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2608](https://ethyca.atlassian.net/browse/PROD-2608) and [PROD-2611](https://ethyca.atlassian.net/browse/PROD-2611)

### Description Of Changes

* fetch “minimal” TCF experience API in privacy center when bundling FidesJS
* modify TCF banner to render only using the minimal API response fields
* remove display of “stacks” entirely, and render purposes & special features from the minimal API

### Code Changes

* [ ] _list your code changes here_

### Steps to Confirm

* Full regression test on FidesJS is recommended.
* No real change to functionality other than the speed in which the banner appears.
* Be sure to check TCF banner UI in both desktop and mobile views.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] if there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* If there are any database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!

[PROD-2608]: https://ethyca.atlassian.net/browse/PROD-2608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PROD-2611]: https://ethyca.atlassian.net/browse/PROD-2611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/fides-js/src/services/api.ts [hunk 1]
```diff
@@ -85,13 +111,23 @@ export const fetchExperience = async (
...
-    return experience;
+    // returning empty obj instead of undefined ensures we can properly cache on
+    // server-side for locations that have no relevant experiences.
+    const experience:
+      | PrivacyExperience
+      | PrivacyExperienceMinimal
+      | EmptyExperience = (body.items && body.items[0]) ?? {};
+
+    const firstLanguage =
+      experience.experience_config?.translations?.[0].language;
```
Comment (eastandwestwind) [inline clients/fides-js/src/services/api.ts:125]: 👍

File: clients/privacy-center/pages/api/fides-js.ts [hunk 2]
```diff
@@ -157,18 +166,29 @@ export default async function handler(
     const fidesRegionString = constructFidesRegionString(geolocation);
 
     if (fidesRegionString) {
+      // Check for a provided "fides_locale" query param or cookie. If present, use it as
+      // the user's preferred language, otherwise use the "accept-language" header
+      // provided by the browser. If all else fails, use the default.
+      const fidesLocale =
+        (req.query.fides_locale as string) || req.cookies?.fides_locale;
```
Comment (eastandwestwind) [inline clients/privacy-center/pages/api/fides-js.ts:173]: do we set `fides_locale` anywhere across our ecosystem? Who uses this that we know of?

File: clients/privacy-center/pages/api/fides-js.ts [hunk 3]
```diff
@@ -157,18 +166,29 @@ export default async function handler(
...
+        userLanguageString,
+        fidesApiUrl:
+          serverSettings.SERVER_SIDE_FIDES_API_URL ||
           environment.settings.FIDES_API_URL,
-        environment.settings.DEBUG,
-        null,
+        debug: environment.settings.DEBUG,
         propertyId,
-      );
+        minimalTCF: true,
```
Comment (eastandwestwind) [inline clients/privacy-center/pages/api/fides-js.ts:190]: Not sure I like using `minimalTCF` for experience fetch code that could either be TCF or non-TCF. Could we rename to `isMinimalExperience` or `isMinimalTranslation`?

File: clients/privacy-center/pages/api/fides-js.ts [hunk 4]
```diff
@@ -157,18 +166,29 @@ export default async function handler(
     const fidesRegionString = constructFidesRegionString(geolocation);
 
     if (fidesRegionString) {
+      // Check for a provided "fides_locale" query param or cookie. If present, use it as
+      // the user's preferred language, otherwise use the "accept-language" header
+      // provided by the browser. If all else fails, use the default.
+      const fidesLocale =
+        (req.query.fides_locale as string) || req.cookies?.fides_locale;
+      const userLanguageString =
+        fidesLocale || req.headers["accept-language"] || DEFAULT_LOCALE;
```
Comment (eastandwestwind) [inline clients/privacy-center/pages/api/fides-js.ts:175]: what is the default locale set to?

File: clients/fides-js/src/components/fides.css [hunk 5]
```diff
@@ -1118,8 +1100,6 @@ button.fides-banner-button.fides-menu-item:not([aria-pressed="true"]):hover {
   }
 
   div#fides-banner-inner-container {
-    display: flex;
-    flex-direction: column;
```
Comment (eastandwestwind) [inline clients/fides-js/src/components/fides.css:1122]: Can you explain why these CSS changes were needed or generally what these changes do?

We're also making some CSS changes here that we need to eval to see if there are any conflicts- [clients/fides-js-extensions/nytimes/src/ext.css](https://github.com/ethyca/fidesplus/pull/1582/files)

**Review by eastandwestwind — CHANGES_REQUESTED** [review]
Nice work so far @gilluminate - I've added some suggestions / questions but mostly looking for a test that checks:

* banner renders minimal response fields initially
* follow-up API call is made after banner renders for full experience
* banner updates to render full experience response

In addition, I'd like clarification around whether or not this is scoped to just TCF. It seems to me that this minimal API call is made across the board, not just for TCF experiences, right?

File: clients/fides-js/src/components/fides.css (hunk 5 above)
Comment (gilluminate) [reply clients/fides-js/src/components/fides.css:1122]: all CSS changes are being made as part of this layout update ticket https://ethyca.atlassian.net/browse/PROD-2611 which is pretty heavily tied to the other TCF updates so I'm including them in this PR. All of these affect TCF only, and they affect the banner only. No style updates for modal at all and no updates for non-TCF banner.

File: clients/privacy-center/pages/api/fides-js.ts (hunk 2 above)
Comment (gilluminate) [reply clients/privacy-center/pages/api/fides-js.ts:173]: I believe MediaVine is using it on a few of their sites that are language specific. Is there a specific concern you are considering here?

File: clients/privacy-center/pages/api/fides-js.ts (hunk 4 above)
Comment (gilluminate) [reply clients/privacy-center/pages/api/fides-js.ts:175]: default is "en"

**Comment by gilluminate** [comment]
> In addition, I'd like clarification around whether or not this is scoped to just TCF. It seems to me that this minimal API call is made across the board, not just for TCF experiences, right?

@eastandwestwind Since we don't know what the experience will be when the initial call is made, so we do supply the minimal request to the api endpoint with the understanding that _if_ TCF is being returned, we want the minimal version. It will be ignored otherwise.

File: clients/privacy-center/pages/api/fides-js.ts (hunk 2 above)
Comment (eastandwestwind) [reply clients/privacy-center/pages/api/fides-js.ts:173]: nope, just curious about the wider implications across the ecosystem ✔️

**Pull request merged**